tests/python/gaia-ui-tests/gaiatest/apps/search/app.py

```diff
@@ -33,15 +40,9 @@ def go_to_url(self, url):
 
     @property
     def history_items_count(self):
+        Wait(self.marionette).until(expected.element_present(*self._history_locator))
         return len(self.marionette.find_elements(*self._history_item_locator))
 
     def wait_for_history_to_load(self, number_of_items=1):
+        Wait(self.marionette).until(expected.element_present(*self._history_locator))
```

Comment:
If for some reason the browser isn't there at all, then wait_for_history_to_load(number_of_items=0) will pass without this change.
I've encountered this (not with wait_for_history_to_load, but with history_items_count)